lib/streamlit/elements/widgets/button_group.py

```diff
@@ -822,10 +822,12 @@ def _transformed_format_func(option: V) -> ButtonGroupProto.Option:
             if len(transformed_parts) > 0:
                 maybe_icon = transformed_parts[0].strip()
                 try:
-                    # we only want to extract material icons because we treat them
```

Comment:
What was the reason to not treat emojis as icons?

Reply:
I am not sure why, but this is based on the comment below
```
transformed = " ".join(transformed_parts[1:])
```
happened previously only for material icon cases, but not for emoji (`transformed_parts` are after applying format_func)

Reply:
I chatted with @raethlein and concluded that it is probably OK to merge it this way. There probably where some concerns around the styling / spacing.. but merging this makes it more consistent with how its treated in our buttons.